src/dvc_data/hashfile/hash.py

```diff
@@ -167,6 +171,7 @@ def hash_file(
     state: "StateBase" = None,
     callback: "Callback" = None,
     info: Optional[dict] = None,
+    text: Optional[bool] = None,
```

Comment:
Unsure if we want to change the dvc-data default for this to be `False` (disable dos2unix by default) instead of `None` (auto-determine based in `istextfile`)

I'd prefer making it default to `False` in dvc-data and pass the explicit `None` from DVC in the event we actually want the legacy istextfile behavior